File: javascript-programming/examples/intro/shift_unshift.js
"use strict";

var names = [ "foo", "bar", "moo", "qux" ];

var first = names.shift();
console.log(first); //  "foo"
console.log(names); // [ "bar", "moo", "qux" ]

names.unshift("boo");
console.log(names); // [ "boo", "bar", "moo", "qux" ]

File: books/perl/src/examples/perlarrays/shift_unshift.pl
#!/usr/bin/perl
use strict;
use warnings;

my @names = ("Foo", "Bar", "Baz");
my $first = shift @names;

print "$first\n";              # Foo
print "@names\n";              # Bar Baz



unshift @names, "Moo";
print "@names\n";              # Moo Bar Baz

